### ISSUE TITLE
Offline mode

### ISSUE THREAD

**Issue by AzzieDev** [SIMKL/script.simkl#20]
The scrobbler gives an error message when it fails because of a lack of internet connection.

It would be great if it could say that it failed and is also added to a queue of items that need to be scrobbled when the internet is reconnected. When it later tries and succeeds to scrobble or check in, it should send queue of other things to be scrobbled too.

**Comment by daviddavo** [comment]
I fixed the error message. But the queue function is going to be implemented as 'syncing library'.

**Comment by kat953162** [comment]
Thats great! Though I hope the 'Syncing Library' keeps the queue if Kodi is closed and reopened.

**Comment by daviddavo** [comment]
You can test it on e4c4cf5cedf59d1b2280b559f75532191517979a
But I won't close this issue because it doesn't detect differences in time. I mean, If you mark as watched something in Kodi it will be marked as watched on Simkl. But if you then "unwatch" it on Simkl, it won't be unwatched on Kodi, so when you sync it will be marked as watched again on Simkl.

**Comment by kat953162** [comment]
That sounds okay for now, but not for the long run. 

Can I have a link to a zip to install?

**Comment by kat953162** [comment]
One issue I realized... I delete watched episodes as soon as they are watched. Wouldn't these be removed from Kodi library, and therefore non synced?

**Comment by daviddavo** [comment]
Yes, they need to stay in the library to be synced

**Comment by kat953162** [comment]
Update to this: It should ping simkl or google to check online status.
Currently, if Kodi is loaded when no internet connection is available, and the internet is restored, no scrobbling will occur unless Kodi is restarted again.

**Comment by ekleop** [comment]
yes, because it tries to sign in on kodi`s start. We should add connection state checking and try to log in once more when connection will be available
